api/v1alpha2/gcpmachine_types.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api/errors"
+)
+
+const (
+	// MachineFinalizer allows ReconcileGCPMachine to clean up GCP resources associated with GCPMachine before
+	// removing it from the apiserver.
+	MachineFinalizer = "gcpmachine.infrastructure.cluster.x-k8s.io"
+)
+
+// GCPMachineSpec defines the desired state of GCPMachine
+type GCPMachineSpec struct {
+	// Zone is references the GCP zone to use for this instance.
+	Zone string `json:"zone"`
+
+	// InstanceType is the type of instance to create. Example: n1.standard-2
+	InstanceType string `json:"instanceType"`
+
+	// Subnet is a reference to the subnetwork to use for this instance. If not specified,
+	// the first subnetwork retrieved from the Cluster Region and Network is picked.
```

Comment:
Again, we probably inherited this, but making this optional feels risky

Reply:
It seems that the default behavior I've seen in GCP when you create an instance is to create the network interface in the specified network (fix incoming for that) and pick a subnet from the zone specified. 

Do we prefer to make both Zone and Subnet required or only one of them?